Should not be able to panic the database by using particular field keys

Hi,
I was trying to include field named "time" which is similar to the predefined "time" field.
While doing so, it didn't gave any error. But while querying the "time" from the measurement, it gave error and after that no single query is executing, even "select" or "show measurements" not working. 
So, I need to go to the /etc/init.d/ and restart the service.
Can anyone explain this issue?

For the record, the panic that is occurring is as follows:
```
panic: interface conversion: interface is string, not map[string]interface {}

goroutine 71 [running]:
github.com/influxdb/influxdb/influxql.(*MapReduceJob).processRawResults(0xc208011f10, 0xc208178158, 0x1, 0x1, 0x1)
        /home/philip/repos/influxdb/src/github.com/influxdb/influxdb/influxql/engine.go:745 +0xa64
github.com/influxdb/influxdb/influxql.(*MapReduceJob).processRawQuery(0xc208011f10, 0xc2081746c0, 0x0)
        /home/philip/repos/influxdb/src/github.com/influxdb/influxdb/influxql/engine.go:355 +0xa39
github.com/influxdb/influxdb/influxql.(*MapReduceJob).Execute(0xc208011f10, 0xc2081746c0, 0xc208133f00)
        /home/philip/repos/influxdb/src/github.com/influxdb/influxdb/influxql/engine.go:81 +0x11ec
github.com/influxdb/influxdb/influxql.(*Executor).execute(0xc2080e44c0, 0xc2081746c0)
        /home/philip/repos/influxdb/src/github.com/influxdb/influxdb/influxql/engine.go:926 +0xba
created by github.com/influxdb/influxdb/influxql.(*Executor).Execute
        /home/philip/repos/influxdb/src/github.com/influxdb/influxdb/influxql/engine.go:905 +0x5a
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -82,6 +82,7 @@ Please see the *Features* section below for full details.
 - [#3736](https://github.com/influxdb/influxdb/pull/3736): Update shard group duration with retention policy changes. Thanks for the report @papylhomme
 - [#3539](https://github.com/influxdb/influxdb/issues/3539): parser incorrectly accepts NaN as numerical value, but not always
 - [#3790](https://github.com/influxdb/influxdb/pull/3790): Fix line protocol parsing equals in measurements and NaN values
+- [#3778](https://github.com/influxdb/influxdb/pull/3778): Don't panic if SELECT on time.
 
 ## v0.9.2 [2015-07-24]
 

--- a/cmd/influxd/run/server_test.go
+++ b/cmd/influxd/run/server_test.go
@@ -1394,6 +1394,11 @@ func TestServer_Query_Common(t *testing.T) {
 			exp:     fmt.Sprintf(`{"results":[{"series":[{"name":"cpu","columns":["time","value"],"values":[["%s",1]]}]}]}`, now.Format(time.RFC3339Nano)),
 		},
 		&Query{
+			name:    "explicitly selecting time and a valid measurement and field should succeed",
+			command: `SELECT time,value FROM db0.rp0.cpu`,
+			exp:     fmt.Sprintf(`{"results":[{"series":[{"name":"cpu","columns":["time","value"],"values":[["%s",1]]}]}]}`, now.Format(time.RFC3339Nano)),
+		},
+		&Query{
 			name:    "selecting a measurement that doesn't exist should result in empty set",
 			command: `SELECT value FROM db0.rp0.idontexist`,
 			exp:     `{"results":[{}]}`,

--- a/influxql/ast.go
+++ b/influxql/ast.go
@@ -1013,10 +1013,9 @@ func (s *SelectStatement) validate(tr targetRequirement) error {
 }
 
 func (s *SelectStatement) validateFields() error {
-	for _, f := range s.NamesInSelect() {
-		if f == "time" {
-			return fmt.Errorf("'time' is an invalid field for SELECT")
-		}
+	ns := s.NamesInSelect()
+	if len(ns) == 1 && ns[0] == "time" {
+		return fmt.Errorf("at least 1 non-time field must be queried")
 	}
 	return nil
 }

--- a/influxql/ast.go
+++ b/influxql/ast.go
@@ -985,6 +985,10 @@ func (s *SelectStatement) hasTimeDimensions(node Node) bool {
 }
 
 func (s *SelectStatement) validate(tr targetRequirement) error {
+	if err := s.validateFields(); err != nil {
+		return err
+	}
+
 	if err := s.validateDistinct(); err != nil {
 		return err
 	}
@@ -1005,6 +1009,15 @@ func (s *SelectStatement) validate(tr targetRequirement) error {
 		return err
 	}
 
+	return nil
+}
+
+func (s *SelectStatement) validateFields() error {
+	for _, f := range s.NamesInSelect() {
+		if f == "time" {
+			return fmt.Errorf("'time' is an invalid field for SELECT")
+		}
+	}
 	return nil
 }
 

--- a/influxql/parser_test.go
+++ b/influxql/parser_test.go
@@ -1206,7 +1206,7 @@ func TestParser_ParseStatement(t *testing.T) {
 		// Errors
 		{s: ``, err: `found EOF, expected SELECT, DELETE, SHOW, CREATE, DROP, GRANT, REVOKE, ALTER, SET at line 1, char 1`},
 		{s: `SELECT`, err: `found EOF, expected identifier, string, number, bool at line 1, char 8`},
-		{s: `SELECT time FROM myserie`, err: `'time' is an invalid field for SELECT`},
+		{s: `SELECT time FROM myseries`, err: `at least 1 non-time field must be queried`},
 		{s: `blah blah`, err: `found blah, expected SELECT, DELETE, SHOW, CREATE, DROP, GRANT, REVOKE, ALTER, SET at line 1, char 1`},
 		{s: `SELECT field1 X`, err: `found X, expected FROM at line 1, char 15`},
 		{s: `SELECT field1 FROM "series" WHERE X +;`, err: `found ;, expected identifier, string, number, bool at line 1, char 38`},

--- a/influxql/parser_test.go
+++ b/influxql/parser_test.go
@@ -1206,6 +1206,7 @@ func TestParser_ParseStatement(t *testing.T) {
 		// Errors
 		{s: ``, err: `found EOF, expected SELECT, DELETE, SHOW, CREATE, DROP, GRANT, REVOKE, ALTER, SET at line 1, char 1`},
 		{s: `SELECT`, err: `found EOF, expected identifier, string, number, bool at line 1, char 8`},
+		{s: `SELECT time FROM myserie`, err: `'time' is an invalid field for SELECT`},
 		{s: `blah blah`, err: `found blah, expected SELECT, DELETE, SHOW, CREATE, DROP, GRANT, REVOKE, ALTER, SET at line 1, char 1`},
 		{s: `SELECT field1 X`, err: `found X, expected FROM at line 1, char 15`},
 		{s: `SELECT field1 FROM "series" WHERE X +;`, err: `found ;, expected identifier, string, number, bool at line 1, char 38`},

